packages/docusaurus-theme-classic/src/include-additional-languages.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import Prism from 'prism-react-renderer/prism';
+import ExecutionEnvironment from '@docusaurus/ExecutionEnvironment';
+import siteConfig from '@generated/docusaurus.config';
+
+if (ExecutionEnvironment.canUseDOM) {
+  const {
+    themeConfig: {prism: {additionalLanguages = []} = {}},
+  } = siteConfig;
+
+  window.Prism = Prism;
+
+  additionalLanguages.forEach(lang => {
+    require(`prismjs/components/prism-${lang}`); // eslint-disable-line
```

Comment:
I might be wrong, but if there's no some context replacement I'm missing, all files matching this query would be included in a bundle

Reply:
Yeah agreed, I think so too! One alternative is to generate a file in one of the plugin steps that generates the list of `require()` statements containing just the languages needed and have this file import it.